src/mesh/rectilineargrid.jl

```diff
@@ -35,12 +35,13 @@ vertex(g::RectilinearGrid{Dim}, ijk::Dims{Dim}) where {Dim} = Point(getindex.(g.
 
 xyz(g::RectilinearGrid) = g.xyz
 
-@generated function XYZ(g::RectilinearGrid{Dim,T}) where {Dim,T}
+XYZ(g::RectilinearGrid{Dim,T}) where {Dim,T} = xyzXYZ(T, xyz(g))
+
+@generated function xyzXYZ(xyz::NTuple{Dim,<:AbstractVector{T}}) where {Dim,T}
```

Comment:
Now that the function accepts a NTuple, we can rename it to `XYZ` and rely on dispatch

Reply:
That means that we will have XYZ with `Grid` types and with `NTuple` to convert the `xyz` to `XYZ`